<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Link-HD: a versatile framework to explore and integrate heterogeneous data</title>
  <meta name="description" content="Link-HD: a versatile framework to explore and integrate heterogeneous data" />
  <meta name="generator" content="bookdown 0.17 and GitBook 2.6.7" />

  <meta property="og:title" content="Link-HD: a versatile framework to explore and integrate heterogeneous data" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Link-HD: a versatile framework to explore and integrate heterogeneous data" />
  
  
  

<meta name="author" content="Laura M Zingaretti, Yuliaxis Ramayo-Caldas" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  


<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path=""><a href="#Introduction"><i class="fa fa-check"></i>Introduction</a><ul>
<li class="chapter" data-level="" data-path=""><a href="#Methods"><i class="fa fa-check"></i>Methods</a></li>
<li class="chapter" data-level="" data-path=""><a href="#STATIS"><i class="fa fa-check"></i>STATIS: a general overview</a></li>
<li class="chapter" data-level="" data-path=""><a href="#Selecting"><i class="fa fa-check"></i>Selecting variables</a></li>
<li class="chapter" data-level="" data-path=""><a href="#Regression"><i class="fa fa-check"></i>Regression Biplot approach</a></li>
<li class="chapter" data-level="" data-path=""><a href="#Differential"><i class="fa fa-check"></i>Differential</a></li>
<li class="chapter" data-level="" data-path=""><a href="#Usage"><i class="fa fa-check"></i>Usage</a></li>
<li class="chapter" data-level="" data-path=""><a href="#Software"><i class="fa fa-check"></i>Software Installation</a></li>
<li class="chapter" data-level="" data-path=""><a href="#Generalities"><i class="fa fa-check"></i>Generalities</a></li>
<li class="chapter" data-level="" data-path=""><a href="#Examples"><i class="fa fa-check"></i>Examples</a></li>
<li class="chapter" data-level="" data-path=""><a href="#Results"><i class="fa fa-check"></i>Results</a></li>
<li class="chapter" data-level="" data-path=""><a href="#Ruminotypes"><i class="fa fa-check"></i>Ruminotypes analysis</a></li>
<li class="chapter" data-level="" data-path=""><a href="#Data"><i class="fa fa-check"></i>Data Processing</a></li>
<li class="chapter" data-level="" data-path=""><a href="#Getting"><i class="fa fa-check"></i>Getting common structure</a></li>
<li class="chapter" data-level="" data-path=""><a href="#Obtaining"><i class="fa fa-check"></i>Obtaining individual projections</a></li>
<li class="chapter" data-level="" data-path=""><a href="#Relationships"><i class="fa fa-check"></i>Relationships between communities</a></li>
<li class="chapter" data-level="" data-path=""><a href="#Sample"><i class="fa fa-check"></i>Sample stratification in ruminotypes-like clusters</a></li>
<li class="chapter" data-level="" data-path=""><a href="#Variable"><i class="fa fa-check"></i>Variable selection</a></li>
<li class="chapter" data-level="" data-path=""><a href="#Taxonomic"><i class="fa fa-check"></i>Taxonomic aggregation and Taxon Set Enrichment Analysis</a></li>
<li class="chapter" data-level="" data-path=""><a href="#Taxon"><i class="fa fa-check"></i>Taxon set enrichment analysis from a user-defined OTUs’ list</a></li>
<li class="chapter" data-level="" data-path=""><a href="#Final"><i class="fa fa-check"></i>Final Remarks</a></li>
<li class="chapter" data-level="" data-path=""><a href="#Appendix"><i class="fa fa-check"></i>Appendix</a></li>
<li class="chapter" data-level="" data-path=""><a href="#References"><i class="fa fa-check"></i>References</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Link-HD: a versatile framework to explore and integrate heterogeneous data</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">Link-HD: a versatile framework to explore and integrate heterogeneous data</h1>
<p class="author"><em>Laura M Zingaretti, Yuliaxis Ramayo-Caldas</em></p>
<p class="date"><em>September 29th, 2020</em></p>
</div>
<div id="Introduction" class="section level1 unnumbered">
<h1>Introduction</h1>
<p>Recent advances in omics technologies have revolutionized biological research providing vast amounts of heterogeneous data. The drop in ‘omics’ technologies costs enables collection of data from multiple sources; facilitating the study of gene expression, proteins, metagenomics, metabolites, among others and to evaluate their relationship with health and diseases status. Although the integration of these heterogeneous data is not trivial, several statistical methods have been developed to handle these challenges <span class="citation">(Mariette and Villa-Vialaneix 2017; Rohart et al. 2017; Argelaguet et al. 2018)</span>. <strong>STATIS-ACT</strong> <span class="citation">(Plantes 1976; Escoufier, Cazes, and others 1976)</span> is a suitable method for analyzing data arising from several configurations.</p>
<p>STATIS is a multi-way <strong>PCA</strong> extension, which can analyze multiple datasets of variables collected from the same individual or the several individuals on which the same variables were measured <span class="citation">(Lavit et al. 1994)</span>. The method summarizes this information into an optimum space called the compromise. STATIS has some constraint since can only be used with continuous data and the subspace generated by STATIS only can establish relationships between common elements in all datasets, i. e. observations (samples) or variables (genes/transcripts/proteins/metabolites/microbial communities). Neither is it possible to establish relationships between observations and variables, nor can it provide a variable selection strategy.</p>
<p>Here, we present <strong>Link-HD</strong>, an R package to integrate multiple heterogeneous datasets based on STATIS. Our approach is a generalization of multidimensional scaling, which uses distance matrices for numerical, categorical or compositional variables instead of the raw data itself. Furthermore, the methodology was enhanced with Regression Biplot <span class="citation">(Gabriel 1971; Gower and Hand 1995)</span>, differential abundance testing to perform variable selection and to identify OTUs that differ between two or more categories and, importantly we introduce a novel <strong>Hypergeometric Taxa Enrichment (HyTE) test</strong> to analyze whether there are enrichments of genus (families) into the selected taxons.</p>
<div id="Methods" class="section level2 unnumbered">
<h2>Methods</h2>
<p>Our package was designed to integrate multiple communities from a holistic view. It can deal with compositional data, and it includes centered log ratio and cumulative-sum scaling (CSS) <span class="citation">(Paulson et al. 2013)</span> transformations. Link-HD also incorporates variable selection and visualization tools to provide a better insight into the structure of the communities, as well as their relationship with target traits.</p>
<p><img src="Complete_Figure.png" alt="" />
(A) Link-HD pipeline overview. First, raw data are transformed using cumulative sum scaling (CSS) <span class="citation">(Paulson et al. 2013)</span> or centered log ratio transformation (clr) <span class="citation">(Aitchison 1982)</span>. Second, the similarity between datasets is assessed by distance or cross -product matrices. Third, the compromise uses the first eigenvectors from RV (vectorial correlation coefficient) to build the compromise matrix (<span class="math inline">\(W\)</span>). The eigen-decomposition of <span class="math inline">\(W\)</span> allows to obtain the projection of each observation on the common structure. Finally, the relationships between the set of observations and some target variable can be indirectly established using a varible selection method (Biplot, Differential abundance, etc.). (B) Link-HD includes visualization tools to explore relationships between data. In this example from the rumen microbiota it is shown: cross-product matrices relationships (upper left panel), ‘ruminotypes’ from the consensus structure (upper right panel) and the microbiota structure (abundances) in each cluster (lower panel).</p>
</div>
<div id="STATIS" class="section level2 unnumbered">
<h2>STATIS: a general overview</h2>
<p>The main objectives of STATIS-ACT <span class="citation">(Plantes 1976; Escoufier, Cazes, and others 1976)</span> are to compare and analyze the relationships between data that share a common collection of observations, but each may have different variables. The data consists of <span class="math inline">\(k\)</span> matrices <span class="math inline">\(X_{[k]}\)</span> of <span class="math inline">\(n \times j_{[k]}\)</span> , where <span class="math inline">\(n\)</span> is the number of observations and <span class="math inline">\(j_{[k]}\)</span> the number of variables measured in <span class="math inline">\(k-th\)</span> matrix. The idea behind STATIS is to obtain a compromise matrix, which is nothing but linear combination of configuration of matrices derived from the raw data. The analysis of this matrix gives a two/three -dimensional representation (principal axes maps) that can be used to interpret its structure. The methodology is implemented in three main phases [<a href="#fig:Method">2</a>]:</p>
<ul>
<li><p>Inter-structure: It measures the similarity between the <span class="math inline">\(k\)</span> configurations of the matrices. This phase is equivalent to defining a distance among the corresponding scalar product matrices.</p></li>
<li><p>Compromise: It generates an optimal weight coefficient for each <span class="math inline">\(k\)</span> matrix, in order to construct a common structure for all configurations.</p></li>
<li><p>Intra-structure: It is a generalized <strong>PCA</strong> analysis of the structure obtained in the previous stage.</p></li>
</ul>
<p><img src="Method_figure.png" alt="" />
STATIS overview. This figure shows the three steps of the STATIS: Inter-structure, Compromise and Intra-structure. In the inter-structure, the similarity dataset structure is analyzed and each <span class="math inline">\(X_{[k]}\)</span> matrix is then projected onto the first and second components of the <span class="math inline">\(Q\)</span> matrix. The compromise uses the first eigenvectors from <span class="math inline">\(Q\)</span> to build the common configuration <span class="math inline">\(W\)</span> matrix. Finally, the intra-structure step is the eigen-decomposition of <span class="math inline">\(W\)</span> and each observation is projected onto the first and second components of the <span class="math inline">\(A\)</span> matrix.</p>
<p>The mathematics behind the methodology can be described as follows. The first task requires computations of <span class="math inline">\(k\)</span> configurations of symmetrical matrices <span class="math inline">\(W_{[k]}\)</span> for each <span class="math inline">\(X_{[k]}\)</span> matrix of dimension <span class="math inline">\(n \times j_{[k]}\)</span> according to equation (<a href="#eq:inter-structure">(1)</a>):</p>
<p><span class="math display" id="eq:inter-structure">\[\begin{equation}
 \tag{1}
W_{[k]}=X_{[k]}M_{[k]}X^\top_{[k]}
\end{equation}\]</span></p>
<p>where <span class="math inline">\(W_{[k]}\)</span> is a square matrix of size <span class="math inline">\(n\)</span>; <span class="math inline">\(M_{[k]}\)</span> is a diagonal matrix including the variable weights, which is generally the same matrix for all the <span class="math inline">\(k\)</span> matrices <span class="math inline">\(M=M_{[k]}=\frac{1}{j_{[k]}}I_{j_{[k]}}\)</span> where <span class="math inline">\(I_{j_{[k]}}\)</span> is the identity matrix of size <span class="math inline">\(j_{[k]}\)</span>; and <span class="math inline">\(X^\top_{[k]}\)</span> denotes the transpose of <span class="math inline">\(X_{[k]}\)</span> matrix. Then, the similarity between the two configurations <span class="math inline">\(k\)</span> and <span class="math inline">\(k^\prime\)</span> can be established computing the Hilbert-Smith product of equation (<span class="math inline">\(&lt;.&gt;_{HS}\)</span>, <span class="citation">(Robert and Escoufier 1976)</span>) presented in equation <a href="#eq:hs">(2)</a> or in its normalized version <span class="math inline">\(\rho_{k,k^\prime}\)</span>.</p>
<p><span class="math display" id="eq:hs-weigthed" id="eq:hs">\[\begin{eqnarray}
\tag{2}
\langle W_{[k]} \arrowvert W_{[k^\prime]}\rangle_{HS} =
\left\{
  \begin{array}{cl}
    tr(DW_{[k]}DW_{[k^\prime]}) &amp; if~k \neq k^\prime\\
    ||W_{[k]}||^2 &amp; if~k = k^\prime
  \end{array}
\right.\\
\tag{3}
\rho_{k,k^\prime}=\frac{\langle W_{[k]} \arrowvert W_{[k^\prime]} \rangle_{HS}}{\sqrt{\langle W_{[k]} \arrowvert W_{[k]} \rangle_{HS}\langle W_{[k^\prime]} \arrowvert W_{[k^\prime]} \rangle_{HS}}}
\end{eqnarray}\]</span></p>
<p><span class="math inline">\(tr(.)\)</span> is the trace of the argument matrix; <span class="math inline">\(D\)</span> is the square weighted observation matrix of size <span class="math inline">\(n\)</span>, which in general is <span class="math inline">\(D=\frac{1}{n}I_n\)</span>; and <span class="math inline">\(||.||^2\)</span> is the classic Euclidean norm-2 of the argument matrix. In this context, equation <a href="#eq:hs">(2)</a> can be geometrically interpreted as the scalar product between two positive semi-definite matrices, which is proportional to the cosine of the angle between them.</p>
<p>The <em>inter-structure’s</em> objective is to decide whether or not there is a common structure to all representative <span class="math inline">\(W_{[k]}\)</span>’s objects. The scalar products <span class="math inline">\(\rho_{k,k^\prime}\)</span> can be organized in a <span class="math inline">\(k \times k\)</span> positive semi-definite matrix <span class="math inline">\(R\)</span> and their spectral decomposition obtains the similarity analysis among the configurations as a generalized correlation coefficient.</p>
<p>However, the formulation from the <a href="#eq:inter-structure">(1)</a> has some restrictions, i.e., it only can be computed when all observations are continuous. This constraint can be averted by replacing the scalar product <a href="#eq:inter-structure">(1)</a> between observations by a generalized distance cross-product matrix. In Link-HD, we implemented euclidean, canberra, pearson, pearsonabs, spearman, spearmanabs, mahalanobis, jaccard, simple matching and Aitchison distances. Then, each <span class="math inline">\(W_k\)</span> is obtained as follows:</p>
<p><span class="math display" id="eq:H" id="eq:distance">\[\begin{eqnarray}
\tag{4}
W_{k}=\frac{1}{2} H_k D^2 H&#39;_k
\\
\tag{5}
H=I-\frac{1}{n} 11&#39;
\end{eqnarray}\]</span></p>
<p>where <span class="math inline">\(D^2\)</span> in <a href="#eq:distance">(4)</a> is a distance matrix computed from the raw data (<span class="math inline">\(X\)</span>) and <span class="math inline">\(W_k\)</span> is the cross-product transformation of <span class="math inline">\(D\)</span>.</p>
<p><span class="math inline">\(W\)</span>, the matrix that summarize all of <span class="math inline">\(W_{k}\)</span> can be obtained by a weighted mean of the <span class="math inline">\(W_{[k]}\)</span> configurations <span class="citation">(Lavit 1988)</span>:</p>
<p><span class="math display" id="eq:compromise-alpha" id="eq:compromise">\[\begin{eqnarray}
\tag{6}
W=\sum_{k=1}^{K}\alpha_{[k]}W_{[k]}\\
\tag{7}
\alpha=(1^{T}q_{1})^{-1} \times q_{1}
\end{eqnarray}\]</span></p>
<p>where, <span class="math inline">\(\alpha\)</span>, is the vector with optimal weights, obtained by re-scaling these values such that their sum is equal to one; <span class="math inline">\(q_1\)</span>, is the first eigenvector of the <span class="math inline">\(R\)</span> matrix and <span class="math inline">\(\alpha_{[k]}\)</span>, is the weight coefficient for the <span class="math inline">\(k\)</span> configuration.</p>
<p>The <em>compromise</em> is a cross-product matrix and it can be eigen-decomposed. Thus, it is possible to obtain a graphical representation from their Principal Components Analysis (<em>PCA</em>) where the scores are the best common representation for the set of <span class="math inline">\(K\)</span> matrices. Moreover, scores can be used to perform a cluster analysis in the common subspace.</p>
</div>
<div id="Selecting" class="section level2 unnumbered">
<h2>Selecting variables</h2>
</div>
<div id="Regression" class="section level2 unnumbered">
<h2>Regression Biplot approach</h2>
<p>Classical STATIS is neither designed to establish relationships between observations and variables, nor to perform variable selection. However, the general regression biplot formulation from <span class="citation">Gower and Hand (1995)</span> averts these limitations.</p>
<p><span class="math display" id="eq:eq10">\[\begin{equation}
X\approx AB^\top+\varepsilon
\tag{8}
\end{equation}\]</span></p>
<p>The equation <a href="#eq:eq10">(8)</a> can be understood as a multivariate regression of <span class="math inline">\(X\)</span> on the coordinates of rows <span class="math inline">\(A\)</span>, when they are fixed, or a multivariate regression of <span class="math inline">\(X^T\)</span> on the coordinates of the variables <span class="math inline">\(B\)</span>, if they are fixed. Then, we could easily define a regression biplot through a classical linear model, i. e., <span class="math inline">\(E(X)=AB^\top\)</span> if the response has a normal distribution, or using a generalized linear model <span class="math inline">\(E(X)=g(\mu)=AB^\top\)</span> when the response variable has an exponential family distribution and the link function used is <span class="math inline">\(g(.)\)</span> <span class="citation">(Greenacre 2010)</span>. Here, we used the ideas from <a href="#eq:eq10">(8)</a> to project all variables into the common subspace obtained by the interstructure step. Variables are selected using the accuracy of the regression biplot, which is measured using the proportion of explained variance by each regression (adjusted r squared from the model).</p>
</div>
<div id="Differential" class="section level2 unnumbered">
<h2>Differential</h2>
<p>Link-HD also implements a differential abundance analysis, which is computed using a rank-based nonparametric Kruskal-Wallis test <span class="citation">(Kruskal and Wallis 1952)</span>. Variables are selected after applying a procedure that controls the false discovery rate (FDR) <span class="citation">(Benjamini and Hochberg 1995)</span>. Note that, more robust approaches like the <span class="citation">Paulson et al. (2013)</span> can be easily implemented.</p>
<p>## Hype (Hypergeometric Taxa Enrichment test) {-#Hype}</p>
<p>Gene set enrichment (GSE) analysis summarizes the information from gene expression into meaningful pathways, supporting biological interpretability. We extended the idea behind GSE to metagenomic by developing an enrichment analysis test that assesses whether a familiy/genus is over-represented from a list of selected OTUs. The method calculates a hypergeometric test (i.e. a test used when sampling without replacement) to answer the question about the number of OTUs of a given family/gender selected by Differential abundance or Biplot are over-represented with respect to all the OTUs of that family/gender that were measured on the samples. The test yields <span class="math inline">\(T=-log(p)\)</span>, where <span class="math inline">\(p\)</span> is the probability that the OTUs from of a certain gender/family were selected by chance, i.e. a higher <span class="math inline">\(T\)</span> value indicates that the given genus/family is enriched.</p>
</div>
<div id="Usage" class="section level2 unnumbered">
<h2>Usage</h2>
</div>
<div id="Software" class="section level2 unnumbered">
<h2>Software Installation</h2>
<p>devtools package is required to install LinkHD</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="co"># if you don&#39;t have devtools, please install it:</span></a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw">install.packages</span>(<span class="st">&quot;devtools&quot;</span>)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="co"># if devtools is already installed, ommit previous step</span></a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="kw">library</span>(devtools)</a>
<a class="sourceLine" id="cb1-5" data-line-number="5">devtools<span class="op">::</span><span class="kw">install_github</span>(<span class="dt">repo =</span> <span class="st">&quot;lauzingaretti/LinkHD&quot;</span>, <span class="dt">dependencies =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb1-6" data-line-number="6"><span class="kw">library</span>(LinkHD)</a></code></pre></div>
<p>Congrats! The software is ready to use!</p>
<p>You can access to the software help by typing the following command:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="st">`</span><span class="dt">?</span><span class="st">`</span>(<span class="st">`</span><span class="dt">?</span><span class="st">`</span>(<span class="kw">LinKData</span>()))</a></code></pre></div>
</div>
<div id="Generalities" class="section level2 unnumbered">
<h2>Generalities</h2>
<p>The software requires a list of data frames as input with the common elements in rows. We implemented the <code>Read_Data</code> function to facilitate data reading and manipulation. <code>Read_Data</code> only requires as argument the path to the parent folder containing all the individual data to be analyzed. Please check the function help for more information.
LinkHD also supports a list of <code>data.frame</code> as input.</p>
<p><strong><em>Warning!</em></strong> Please remember to name the list using the input data names and ceck that all the data has the same rownames..</p>
</div>
<div id="Examples" class="section level2 unnumbered">
<h2>Examples</h2>
<p>Public data from the TARA Ocean expedition (<a href="https://oceans.taraexpeditions.org/en/m/about-tara/les-expeditions/tara-oceans/" class="uri">https://oceans.taraexpeditions.org/en/m/about-tara/les-expeditions/tara-oceans/</a>), as well as data from the ruminal metataxonomic communities (including bacteria, archaea and protozoa data) are used to illustrate our methodology.</p>
</div>
<div id="Results" class="section level2 unnumbered">
<h2>Results</h2>
<p>TARA ocean dataset contains environmental, structural and funcional information from prokaryotic communities. As in the mixKernel analysis <span class="citation">(Mariette and Villa-Vialaneix 2017)</span>, <strong>Link-HD</strong> shows the relevant role of temperature of the Proteobacteria phyla on the structure of this ecosystem I. The following lines show how to analyse TaraOcean dataset with <strong>Link-HD</strong> package.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw">library</span>(LinkHD)</a></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="kw">data</span>(<span class="st">&quot;Taraoceans&quot;</span>)</a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="kw">data</span>(Taraoceans)</a>
<a class="sourceLine" id="cb4-3" data-line-number="3">pro.phylo &lt;-<span class="st"> </span>Taraoceans<span class="op">$</span>taxonomy[, <span class="st">&quot;Phylum&quot;</span>]</a>
<a class="sourceLine" id="cb4-4" data-line-number="4">TaraOc &lt;-<span class="st"> </span><span class="kw">list</span>(Taraoceans<span class="op">$</span>phychem, <span class="kw">as.data.frame</span>(Taraoceans<span class="op">$</span>pro.phylo), <span class="kw">as.data.frame</span>(Taraoceans<span class="op">$</span>pro.NOGs))</a>
<a class="sourceLine" id="cb4-5" data-line-number="5">TaraOc_<span class="dv">1</span> &lt;-<span class="st"> </span><span class="kw">scale</span>(TaraOc[[<span class="dv">1</span>]])</a>
<a class="sourceLine" id="cb4-6" data-line-number="6">Normalization &lt;-<span class="st"> </span><span class="kw">lapply</span>(<span class="kw">list</span>(TaraOc[[<span class="dv">2</span>]], TaraOc[[<span class="dv">3</span>]]), <span class="cf">function</span>(x) {</a>
<a class="sourceLine" id="cb4-7" data-line-number="7">    <span class="kw">DataProcessing</span>(x, <span class="dt">Method =</span> <span class="st">&quot;Compositional&quot;</span>)</a>
<a class="sourceLine" id="cb4-8" data-line-number="8">})</a>
<a class="sourceLine" id="cb4-9" data-line-number="9"><span class="kw">colnames</span>(Normalization[[<span class="dv">1</span>]]) =<span class="st"> </span>pro.phylo</a>
<a class="sourceLine" id="cb4-10" data-line-number="10"><span class="kw">colnames</span>(Normalization[[<span class="dv">2</span>]]) =<span class="st"> </span>Taraoceans<span class="op">$</span>GO</a>
<a class="sourceLine" id="cb4-11" data-line-number="11">TaraOc &lt;-<span class="st"> </span><span class="kw">list</span>(TaraOc_<span class="dv">1</span>, Normalization[[<span class="dv">1</span>]], Normalization[[<span class="dv">2</span>]])</a>
<a class="sourceLine" id="cb4-12" data-line-number="12"><span class="kw">names</span>(TaraOc) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;phychem&quot;</span>, <span class="st">&quot;pro_phylo&quot;</span>, <span class="st">&quot;pro_NOGs&quot;</span>)</a>
<a class="sourceLine" id="cb4-13" data-line-number="13">TaraOc &lt;-<span class="st"> </span><span class="kw">lapply</span>(TaraOc, as.data.frame)</a>
<a class="sourceLine" id="cb4-14" data-line-number="14">TaraOc &lt;-<span class="st"> </span><span class="kw">lapply</span>(TaraOc, as.data.frame)</a>
<a class="sourceLine" id="cb4-15" data-line-number="15">Tara_Out &lt;-<span class="st"> </span><span class="kw">LinkData</span>(TaraOc, <span class="dt">Scale =</span> <span class="ot">FALSE</span>, <span class="dt">Center =</span> <span class="ot">FALSE</span>, <span class="dt">Distance =</span> <span class="kw">c</span>(<span class="st">&quot;ScalarProduct&quot;</span>, </a>
<a class="sourceLine" id="cb4-16" data-line-number="16">    <span class="st">&quot;Euclidean&quot;</span>, <span class="st">&quot;Euclidean&quot;</span>))</a></code></pre></div>
<p>The compromise plot from Tara Oceans can be obtained by the next function:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="kw">CompromisePlot</span>(Tara_Out) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">panel.grid.major =</span> <span class="kw">element_blank</span>(), <span class="dt">panel.grid.minor =</span> <span class="kw">element_blank</span>(), </a>
<a class="sourceLine" id="cb5-2" data-line-number="2">    <span class="dt">panel.background =</span> <span class="kw">element_blank</span>(), <span class="dt">axis.line =</span> <span class="kw">element_line</span>(<span class="dt">colour =</span> <span class="st">&quot;black&quot;</span>))</a></code></pre></div>
<p><img src="LinkHD_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
<p>The correlation plot between tables for the Tara Oceans data can be easily obtained by:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="kw">CorrelationPlot</span>(Tara_Out) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">panel.grid.major =</span> <span class="kw">element_blank</span>(), <span class="dt">panel.grid.minor =</span> <span class="kw">element_blank</span>(), </a>
<a class="sourceLine" id="cb6-2" data-line-number="2">    <span class="dt">panel.background =</span> <span class="kw">element_blank</span>(), <span class="dt">axis.line =</span> <span class="kw">element_line</span>(<span class="dt">colour =</span> <span class="st">&quot;black&quot;</span>))</a></code></pre></div>
<p><img src="LinkHD_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
<p>The correlation between pro_NOGs and pro_phylo is higher than with phychemestry variables.</p>
<p>We select variables using Biplot methodology</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">Selection &lt;-<span class="st"> </span><span class="kw">VarSelection</span>(Tara_Out, TaraOc, <span class="dt">Crit =</span> <span class="st">&quot;Rsquare&quot;</span>, <span class="dt">perc =</span> <span class="fl">0.95</span>)</a></code></pre></div>
<p>The following plot just shows the frequency of selected variables from all the tables.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="kw">library</span>(reshape2)</a>
<a class="sourceLine" id="cb8-2" data-line-number="2">Ds &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">table</span>(<span class="kw">Variables</span>(Selection)))</a>
<a class="sourceLine" id="cb8-3" data-line-number="3"><span class="kw">ggplot</span>(Ds, <span class="kw">aes</span>(<span class="dt">x =</span> Var1, <span class="dt">y =</span> Freq)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_bar</span>(<span class="dt">stat =</span> <span class="st">&quot;identity&quot;</span>, <span class="dt">fill =</span> <span class="st">&quot;steelblue&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb8-4" data-line-number="4"><span class="st">    </span>ggpubr<span class="op">::</span><span class="kw">rotate_x_text</span>(<span class="dv">45</span>)</a></code></pre></div>
<p><img src="LinkHD_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
<p>Additional analysis and examples about Tara Ocean can be found in the <strong>Link-HD</strong> manual. Now, we will move to the Ruminotypes example.</p>
</div>
<div id="Ruminotypes" class="section level2 unnumbered">
<h2>Ruminotypes analysis</h2>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="kw">library</span>(LinkHD)</a>
<a class="sourceLine" id="cb9-2" data-line-number="2"><span class="kw">data</span>(<span class="st">&quot;Ruminotypes&quot;</span>)</a></code></pre></div>
<p>In this example, we will analyze metataxonomic data of ruminal communities (including bacteria, archaea and protozoa data) from 65 Holstein cows and we will be assesing the relationship between communities with methane emmision yield (<span class="math inline">\(CH_{4}y\)</span>, g <span class="math inline">\(CH_{4}\)</span>/kg feed intake).</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1"><span class="kw">library</span>(<span class="st">&quot;LinkHD&quot;</span>)</a>
<a class="sourceLine" id="cb10-2" data-line-number="2"><span class="kw">library</span>(<span class="st">&quot;ggplot2&quot;</span>)</a>
<a class="sourceLine" id="cb10-3" data-line-number="3"><span class="kw">options</span>(<span class="dt">width =</span> <span class="dv">50</span>)</a></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="kw">data</span>(<span class="st">&quot;Ruminotypes&quot;</span>)</a></code></pre></div>
</div>
<div id="Data" class="section level2 unnumbered">
<h2>Data Processing</h2>
<p>The ´´´DataProcessing´´´ function has two arguments, the data (data.frame) to be analized and the pre-processing method to be used. For continous data, we recommend the “Standard” method. Two alternative options are available for microbiome data: a centered log-ratios transformation (clr) <span class="citation">(Aitchison 1982)</span> is applied using the “Compositional” option, while “TSS” produces the Total Sum Scaling transformation. This step is alternative and you can use both, a custom function, as well as, a transformation method from another package.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1">Normalization &lt;-<span class="st"> </span><span class="kw">lapply</span>(<span class="kw">list</span>(Ruminotypes<span class="op">$</span><span class="st">`</span><span class="dt">16_S</span><span class="st">`</span>, Ruminotypes<span class="op">$</span>Archaea, </a>
<a class="sourceLine" id="cb12-2" data-line-number="2">    Ruminotypes<span class="op">$</span><span class="st">`</span><span class="dt">18_S</span><span class="st">`</span>), <span class="cf">function</span>(x) {</a>
<a class="sourceLine" id="cb12-3" data-line-number="3">    <span class="kw">DataProcessing</span>(x, <span class="dt">Method =</span> <span class="st">&quot;Compositional&quot;</span>)</a>
<a class="sourceLine" id="cb12-4" data-line-number="4">})</a>
<a class="sourceLine" id="cb12-5" data-line-number="5">Dataset &lt;-<span class="st"> </span>Normalization</a>
<a class="sourceLine" id="cb12-6" data-line-number="6"><span class="kw">names</span>(Dataset) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;16_S&quot;</span>, <span class="st">&quot;Archaea&quot;</span>, <span class="st">&quot;18_S&quot;</span>)</a></code></pre></div>
<p>Link-HD also allows as input MultiAssayExperiment data, a Bioconductor constructor that combines multiple data elements from different experiment.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="kw">library</span>(MultiAssayExperiment)</a></code></pre></div>
<pre><code>## Loading required package: SummarizedExperiment</code></pre>
<pre><code>## Loading required package: GenomicRanges</code></pre>
<pre><code>## Loading required package: stats4</code></pre>
<pre><code>## Loading required package: BiocGenerics</code></pre>
<pre><code>## Loading required package: parallel</code></pre>
<pre><code>## 
## Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:parallel&#39;:
## 
##     clusterApply, clusterApplyLB,
##     clusterCall, clusterEvalQ,
##     clusterExport, clusterMap, parApply,
##     parCapply, parLapply, parLapplyLB,
##     parRapply, parSapply, parSapplyLB</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     IQR, mad, sd, var, xtabs</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     anyDuplicated, append, as.data.frame,
##     basename, cbind, colnames, dirname,
##     do.call, duplicated, eval, evalq,
##     Filter, Find, get, grep, grepl,
##     intersect, is.unsorted, lapply, Map,
##     mapply, match, mget, order, paste, pmax,
##     pmax.int, pmin, pmin.int, Position,
##     rank, rbind, Reduce, rownames, sapply,
##     setdiff, sort, table, tapply, union,
##     unique, unsplit, which, which.max,
##     which.min</code></pre>
<pre><code>## Loading required package: S4Vectors</code></pre>
<pre><code>## 
## Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:base&#39;:
## 
##     expand.grid</code></pre>
<pre><code>## Loading required package: IRanges</code></pre>
<pre><code>## Loading required package: GenomeInfoDb</code></pre>
<pre><code>## Loading required package: Biobase</code></pre>
<pre><code>## Welcome to Bioconductor
## 
##     Vignettes contain introductory material;
##     view with &#39;browseVignettes()&#39;. To cite
##     Bioconductor, see &#39;citation(&quot;Biobase&quot;)&#39;,
##     and for packages &#39;citation(&quot;pkgname&quot;)&#39;.</code></pre>
<pre><code>## Loading required package: DelayedArray</code></pre>
<pre><code>## Loading required package: matrixStats</code></pre>
<pre><code>## 
## Attaching package: &#39;matrixStats&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:Biobase&#39;:
## 
##     anyMissing, rowMedians</code></pre>
<pre><code>## Loading required package: BiocParallel</code></pre>
<pre><code>## 
## Attaching package: &#39;DelayedArray&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:matrixStats&#39;:
## 
##     colMaxs, colMins, colRanges, rowMaxs,
##     rowMins, rowRanges</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     aperm, apply, rowsum</code></pre>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb38-1" data-line-number="1"><span class="kw">library</span>(LinkHD)</a>
<a class="sourceLine" id="cb38-2" data-line-number="2"><span class="kw">data</span>(<span class="st">&quot;Ruminotypes&quot;</span>)</a>
<a class="sourceLine" id="cb38-3" data-line-number="3">Normalization &lt;-<span class="st"> </span><span class="kw">lapply</span>(<span class="kw">list</span>(Ruminotypes<span class="op">$</span><span class="st">`</span><span class="dt">16_S</span><span class="st">`</span>, Ruminotypes<span class="op">$</span>Archaea, </a>
<a class="sourceLine" id="cb38-4" data-line-number="4">    Ruminotypes<span class="op">$</span><span class="st">`</span><span class="dt">18_S</span><span class="st">`</span>), <span class="cf">function</span>(x) {</a>
<a class="sourceLine" id="cb38-5" data-line-number="5">    <span class="kw">DataProcessing</span>(x, <span class="dt">Method =</span> <span class="st">&quot;Compositional&quot;</span>)</a>
<a class="sourceLine" id="cb38-6" data-line-number="6">})</a>
<a class="sourceLine" id="cb38-7" data-line-number="7">Dataset &lt;-<span class="st"> </span>Normalization</a>
<a class="sourceLine" id="cb38-8" data-line-number="8"><span class="kw">names</span>(Dataset) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;16_S&quot;</span>, <span class="st">&quot;Archaea&quot;</span>, <span class="st">&quot;18_S&quot;</span>)</a>
<a class="sourceLine" id="cb38-9" data-line-number="9">Dataset &lt;-<span class="st"> </span><span class="kw">MultiAssayExperiment</span>(<span class="dt">experiments =</span> Dataset)</a></code></pre></div>
<pre><code>## ExperimentList contains data.frame or DataFrame,
##   potential for errors with mixed data types</code></pre>
</div>
<div id="Getting" class="section level2 unnumbered">
<h2>Getting common structure</h2>
<p><em>LinkData</em> is the main function of Link-HD, which performs the integration step (i.e. the STATIS methodology incorporating distances options). After data preprocessing (centered log ratio, clr transformation in our example),the Euclidean distance is used to assess the similarity (distance) between samples. Note that the combination of <strong>clr</strong> and <strong>Euclidean</strong> is equivalent to applying the Aitchison distance, suited for compositional data analysis <span class="citation">(Aitchison 1982)</span>. Our method also allows Bray-Curtis distance, which is a popular metric in microbial ecology.</p>
<p>Link_Data outputs the structure of all the samples into the compromise. This valuable information can be used to explore the relationship of the samples under study with some external phenotypes, to select variables responsibles of the given layout and to obtain cluster of samples.</p>
<p>Based on preliminary results about the <strong><em>Ruminotypes</em></strong> structure of cows microbiota<span class="citation">(Kittelmann et al. 2013; Danielsson et al. 2017; Ramayo-Caldas et al. 2019)</span>, samples were stratified using the partitioning around medioids (pam) algorithm, through the n_cluster argument.</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb40-1" data-line-number="1">Output &lt;-<span class="st"> </span><span class="kw">LinkData</span>(Dataset, <span class="dt">Distance =</span> <span class="kw">rep</span>(<span class="st">&quot;euclidean&quot;</span>, </a>
<a class="sourceLine" id="cb40-2" data-line-number="2">    <span class="dv">3</span>), <span class="dt">Scale =</span> <span class="ot">FALSE</span>, <span class="dt">Center =</span> <span class="ot">FALSE</span>, <span class="dt">nCluster =</span> <span class="dv">3</span>)</a>
<a class="sourceLine" id="cb40-3" data-line-number="3"><span class="co"># variance explained by compromise coordinates</span></a>
<a class="sourceLine" id="cb40-4" data-line-number="4"><span class="kw">CompromisePlot</span>(Output) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">panel.grid.major =</span> <span class="kw">element_blank</span>(), </a>
<a class="sourceLine" id="cb40-5" data-line-number="5">    <span class="dt">panel.grid.minor =</span> <span class="kw">element_blank</span>(), <span class="dt">panel.background =</span> <span class="kw">element_blank</span>(), </a>
<a class="sourceLine" id="cb40-6" data-line-number="6">    <span class="dt">axis.line =</span> <span class="kw">element_line</span>(<span class="dt">colour =</span> <span class="st">&quot;black&quot;</span>))</a></code></pre></div>
<p><img src="LinkHD_files/figure-html/unnamed-chunk-14-1.png" width="672" /></p>
<p>The color intensity is proportional to the quality of representation of each individual in the two-dimensional space from the compromise PCA (more light blue means that the observation is better represented in a low dimentional space).</p>
<p>Optionally, data could be a list of S4 ExpressionSet objects. The advantage of ExpressionSet is their capability to describe samples and features in the experiment, as well as information related to the whole experiment and data-processing. To use ExpressionSet class, please follow the next lines:</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb41-1" data-line-number="1">B &lt;-<span class="st"> </span><span class="kw">ExpressionSet</span>()</a>
<a class="sourceLine" id="cb41-2" data-line-number="2">Dataset1 &lt;-<span class="st"> </span>Normalization</a>
<a class="sourceLine" id="cb41-3" data-line-number="3">A &lt;-<span class="st"> </span><span class="kw">lapply</span>(Dataset1, as.matrix)</a>
<a class="sourceLine" id="cb41-4" data-line-number="4">B &lt;-<span class="st"> </span><span class="kw">lapply</span>(A, ExpressionSet)</a>
<a class="sourceLine" id="cb41-5" data-line-number="5">Output_ &lt;-<span class="st"> </span><span class="kw">LinkData</span>(B, <span class="dt">Distance =</span> <span class="kw">rep</span>(<span class="st">&quot;euclidean&quot;</span>, </a>
<a class="sourceLine" id="cb41-6" data-line-number="6">    <span class="dv">3</span>), <span class="dt">Scale =</span> <span class="ot">FALSE</span>, <span class="dt">Center =</span> <span class="ot">FALSE</span>, <span class="dt">nCluster =</span> <span class="dv">3</span>)</a></code></pre></div>
</div>
<div id="Obtaining" class="section level2 unnumbered">
<h2>Obtaining individual projections</h2>
<p>The sample distribution within each input dataset (equivalent to coordinates principal analysis) can be explored using the Globalplot option. This plot is the projection of each individual table into the compromise space.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb42-1" data-line-number="1"><span class="kw">GlobalPlot</span>(Output)</a></code></pre></div>
<p><img src="LinkHD_files/figure-html/unnamed-chunk-16-1.png" width="672" /></p>
</div>
<div id="Relationships" class="section level2 unnumbered">
<h2>Relationships between communities</h2>
<p>We use the Generalized Multivariate Correlation Coefficient RV to estimate the relationship between each microbial community. The results suggest a strong relationship between the prokaryotic communities of Archaea and Bacteria (16_S).</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb43-1" data-line-number="1"><span class="kw">CorrelationPlot</span>(Output)</a></code></pre></div>
<p><img src="LinkHD_files/figure-html/unnamed-chunk-17-1.png" width="672" /></p>
</div>
<div id="Sample" class="section level2 unnumbered">
<h2>Sample stratification in ruminotypes-like clusters</h2>
<p>As the original publication that found a ruminotype-like structure for rumen bacteria <span class="citation">(Ramayo-Caldas et al. 2019)</span>, <strong>Link-HD</strong> recovers a Ruminotypes-like cluster structure for the rumen ecosystem.</p>
<p>Note that our function is flexible and the algorithm can be run for different configurations or set numbers of cluster. We have not implemented an algorithm to obtain the optimum number of groups, but it could be easily done using extant R functions, like e.g., the Silhouette coefficient <span class="citation">(Rousseeuw 1987)</span>.</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb44-1" data-line-number="1">mydata =<span class="st"> </span><span class="kw">compromise_coords</span>(Output)</a>
<a class="sourceLine" id="cb44-2" data-line-number="2"><span class="co"># set colors</span></a>
<a class="sourceLine" id="cb44-3" data-line-number="3">cols &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">`</span><span class="dt">1</span><span class="st">`</span> =<span class="st"> &quot;red&quot;</span>, <span class="st">`</span><span class="dt">2</span><span class="st">`</span> =<span class="st"> &quot;blue&quot;</span>, <span class="st">`</span><span class="dt">3</span><span class="st">`</span> =<span class="st"> &quot;orange&quot;</span>)</a>
<a class="sourceLine" id="cb44-4" data-line-number="4">mydata<span class="op">$</span>fit.cluster &lt;-<span class="st"> </span><span class="kw">as.factor</span>(mydata<span class="op">$</span>fit.cluster)</a>
<a class="sourceLine" id="cb44-5" data-line-number="5">p &lt;-<span class="st"> </span><span class="kw">ggplot</span>(mydata, <span class="kw">aes</span>(Dim<span class="fl">.1</span>, Dim<span class="fl">.2</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">colour =</span> fit.cluster))</a>
<a class="sourceLine" id="cb44-6" data-line-number="6">p <span class="op">+</span><span class="st"> </span><span class="kw">scale_colour_manual</span>(<span class="dt">values =</span> cols) <span class="op">+</span><span class="st"> </span><span class="kw">theme_classic</span>()</a></code></pre></div>
<p><img src="LinkHD_files/figure-html/unnamed-chunk-18-1.png" width="672" /></p>
<p>The obtained clusters of samples can be used to evaluate their association with external variables (i.e., host phenotype information). Previous studies <span class="citation">(Kittelmann et al. 2013; Danielsson et al. 2017; Ramayo-Caldas et al. 2019)</span> reported a link between the structure of the rumen microbiome and <span class="math inline">\(CH_4\)</span> emission. Therefore, we contrast the <span class="math inline">\(CH_4y\)</span> emission levels between the three clusters of samples and as in the previous studies, cluster assignation was significantly associated with <span class="math inline">\(CH_4\)</span> emission. Significant differences between two out of three groups were found, cows clustered within C1 and C2 having lower and higher <span class="math inline">\(CH_4\)</span> emission.</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb45-1" data-line-number="1"><span class="kw">library</span>(emmeans)</a>
<a class="sourceLine" id="cb45-2" data-line-number="2">Dat &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">y =</span> Ruminotypes<span class="op">$</span>phenotype<span class="op">$</span>ch4y, <span class="dt">E =</span> <span class="kw">as.factor</span>(mydata<span class="op">$</span>fit.cluster))</a>
<a class="sourceLine" id="cb45-3" data-line-number="3">s =<span class="st"> </span><span class="kw">lm</span>(y <span class="op">~</span><span class="st"> </span>E, <span class="dt">data =</span> Dat)</a>
<a class="sourceLine" id="cb45-4" data-line-number="4"><span class="co"># summary(s)</span></a>
<a class="sourceLine" id="cb45-5" data-line-number="5"><span class="kw">lsmeans</span>(s, <span class="st">&quot;E&quot;</span>)</a></code></pre></div>
<pre><code>##  E lsmean    SE df lower.CL upper.CL
##  1   23.6 0.567 58     22.5     24.7
##  2   26.3 0.688 58     25.0     27.7
##  3   23.1 0.650 58     21.8     24.4
## 
## Confidence level used: 0.95</code></pre>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb47-1" data-line-number="1"><span class="kw">pairs</span>(<span class="kw">lsmeans</span>(s, <span class="st">&quot;E&quot;</span>))</a></code></pre></div>
<pre><code>##  contrast estimate    SE df t.ratio p.value
##  1 - 2      -2.734 0.891 58 -3.068  0.0091 
##  1 - 3       0.485 0.863 58  0.562  0.8405 
##  2 - 3       3.219 0.946 58  3.401  0.0034 
## 
## P value adjustment: tukey method for comparing a family of 3 estimates</code></pre>
</div>
<div id="Variable" class="section level2 unnumbered">
<h2>Variable selection</h2>
<p><strong>Link-HD</strong> offers two alternative methods for variable selection. The first one is based on regression biplot. From a general point of view, a regression biplot can be understood as the decomposition of a target matrix. Therefore, it could be interpreted as a projection of each variable (i.e., OTUs) onto the compromise structure. To accomplish that goal, we used an external linear model to obtain the regressors coefficients (by projecting all OTUs on the n first compromise axis). Finally, we use the proportion of explained variance by each regression (<span class="math inline">\(R^2\)</span>) to select a proportion (user defined) of variables with a higher R-squared. The second approach is based on differential abundance analysis, which is computed through a non-parametric Kruskall Wallis test.</p>
<p>Finally, false discovery rate (FDR) is applied to select only variables with an FDR &lt; 0.05 (this level can also be defined by the user).</p>
<p>The biplot procedure can be implemented using the Select_Var function as folow. The Crit argument refers to the criteria used to select the variables and the perc is to establish the percentage of variables to be selected. Intercept is set to F, i.e. FALSE.</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb49-1" data-line-number="1">Select_Var &lt;-<span class="st"> </span><span class="kw">VarSelection</span>(Output, <span class="dt">Data =</span> Dataset, </a>
<a class="sourceLine" id="cb49-2" data-line-number="2">    <span class="dt">Crit =</span> <span class="st">&quot;Rsquare&quot;</span>, <span class="dt">perc =</span> <span class="fl">0.9</span>)</a></code></pre></div>
<p>The differential abundance procedure is implemented through the dAB function:</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb50-1" data-line-number="1">M &lt;-<span class="st"> </span><span class="kw">dAB</span>(Output, <span class="dt">Data =</span> Dataset)</a></code></pre></div>
<p>Note that other functions from metagenomeSeq package can be used.</p>
</div>
<div id="Taxonomic" class="section level2 unnumbered">
<h2>Taxonomic aggregation and Taxon Set Enrichment Analysis</h2>
<p>The taxonomic classification of the OTUs can be used to aggregate them at genus/family level, which facilitates the interpretation of the results. <code>OTU_2Taxa</code> is a function to answer the question about a group of selected OTUs is ‘enriched’ for a particular taxonomic level (i.e., family or genus). The function requires a list of selected OTUs and their taxonomic information.</p>
<p>The following lines split selected variables from Biplot methodology by each table, i.e., which selected variables come from of Archaea, Bacteria and protozoa.</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb51-1" data-line-number="1"><span class="co"># Select only the selected variables form</span></a>
<a class="sourceLine" id="cb51-2" data-line-number="2"><span class="co"># Select_Var object</span></a>
<a class="sourceLine" id="cb51-3" data-line-number="3">SelectedB_16S &lt;-<span class="st"> </span><span class="kw">Variables</span>(Select_Var)[<span class="kw">VarTable</span>(Select_Var) <span class="op">==</span><span class="st"> </span></a>
<a class="sourceLine" id="cb51-4" data-line-number="4"><span class="st">    &quot;16_S&quot;</span>]</a>
<a class="sourceLine" id="cb51-5" data-line-number="5">SelectedB_Archaea &lt;-<span class="st"> </span><span class="kw">Variables</span>(Select_Var)[<span class="kw">VarTable</span>(Select_Var) <span class="op">==</span><span class="st"> </span></a>
<a class="sourceLine" id="cb51-6" data-line-number="6"><span class="st">    &quot;Archaea&quot;</span>]</a>
<a class="sourceLine" id="cb51-7" data-line-number="7">SelectedB_18S &lt;-<span class="st"> </span><span class="kw">Variables</span>(Select_Var)[<span class="kw">VarTable</span>(Select_Var) <span class="op">==</span><span class="st"> </span></a>
<a class="sourceLine" id="cb51-8" data-line-number="8"><span class="st">    &quot;18_S&quot;</span>]</a></code></pre></div>
<p>The same as before, but for selected variables from the dAB function:</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb52-1" data-line-number="1">Selec_16S &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(Ruminotypes[[<span class="dv">1</span>]][, <span class="kw">colnames</span>(Dataset[[<span class="dv">1</span>]]) <span class="op">%in%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb52-2" data-line-number="2"><span class="st">    </span><span class="kw">names</span>(M[[<span class="dv">1</span>]])])</a>
<a class="sourceLine" id="cb52-3" data-line-number="3">Selec_Archaea &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(Ruminotypes[[<span class="dv">2</span>]][, <span class="kw">colnames</span>(Dataset[[<span class="dv">2</span>]]) <span class="op">%in%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb52-4" data-line-number="4"><span class="st">    </span><span class="kw">names</span>(M[[<span class="dv">2</span>]])])</a>
<a class="sourceLine" id="cb52-5" data-line-number="5">Selec_18S &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(Ruminotypes[[<span class="dv">3</span>]][, <span class="kw">colnames</span>(Dataset[[<span class="dv">3</span>]]) <span class="op">%in%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb52-6" data-line-number="6"><span class="st">    </span><span class="kw">names</span>(M[[<span class="dv">3</span>]])])</a></code></pre></div>
<p>The following code shows how to run the OTU2Taxa function and how to plot the output. The plot shows the families of bacterial enriched in our analysis.</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb53-1" data-line-number="1"><span class="kw">library</span>(reshape2)</a>
<a class="sourceLine" id="cb53-2" data-line-number="2">SignTaxa &lt;-<span class="st"> </span><span class="kw">OTU2Taxa</span>(<span class="dt">Selection =</span> <span class="kw">VarTable</span>(Select_Var), </a>
<a class="sourceLine" id="cb53-3" data-line-number="3">    <span class="dt">TaxonInfo =</span> Ruminotypes<span class="op">$</span>Taxa_16S, <span class="dt">tableName =</span> <span class="st">&quot;16_S&quot;</span>, </a>
<a class="sourceLine" id="cb53-4" data-line-number="4">    <span class="dt">AnalysisLev =</span> <span class="st">&quot;Family&quot;</span>)</a>
<a class="sourceLine" id="cb53-5" data-line-number="5">melted_Table &lt;-<span class="st"> </span><span class="kw">melt</span>(SignTaxa<span class="op">$</span>TotalUp1)</a>
<a class="sourceLine" id="cb53-6" data-line-number="6"><span class="kw">ggplot</span>(<span class="dt">data =</span> melted_Table, <span class="kw">aes</span>(<span class="dt">x =</span> <span class="kw">reorder</span>(<span class="kw">rownames</span>(melted_Table), </a>
<a class="sourceLine" id="cb53-7" data-line-number="7">    <span class="op">-</span>value), <span class="dt">y =</span> value)) <span class="op">+</span><span class="st"> </span><span class="kw">scale_fill_gradient</span>(<span class="dt">low =</span> <span class="st">&quot;steelblue&quot;</span>, </a>
<a class="sourceLine" id="cb53-8" data-line-number="8">    <span class="dt">high =</span> <span class="st">&quot;#63D2E4&quot;</span>, <span class="dt">na.value =</span> <span class="st">&quot;steelblue&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">geom_bar</span>(<span class="dt">stat =</span> <span class="st">&quot;identity&quot;</span>, </a>
<a class="sourceLine" id="cb53-9" data-line-number="9">    <span class="dt">fill =</span> <span class="st">&quot;#63D2E4&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_classic</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.ticks =</span> <span class="kw">element_blank</span>(), </a>
<a class="sourceLine" id="cb53-10" data-line-number="10">    <span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>(), <span class="dt">axis.title.y =</span> <span class="kw">element_blank</span>()) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb53-11" data-line-number="11"><span class="st">    </span>ggpubr<span class="op">::</span><span class="kw">rotate_x_text</span>(<span class="dv">90</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="kw">rel</span>(<span class="fl">1.5</span>)))</a></code></pre></div>
<p><img src="LinkHD_files/figure-html/unnamed-chunk-24-1.png" width="672" /></p>
<p>The following figure shows the Bacterial enriched from the differential abundance analysis.</p>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb54-1" data-line-number="1">SignTaxa &lt;-<span class="st"> </span><span class="kw">OTU2Taxa</span>(<span class="dt">Selection =</span> M, <span class="dt">TaxonInfo =</span> Ruminotypes<span class="op">$</span>Taxa_16S, </a>
<a class="sourceLine" id="cb54-2" data-line-number="2">    <span class="dt">tableName =</span> <span class="st">&quot;16_S&quot;</span>, <span class="dt">AnalysisLev =</span> <span class="st">&quot;Family&quot;</span>)</a>
<a class="sourceLine" id="cb54-3" data-line-number="3">melted_Table &lt;-<span class="st"> </span><span class="kw">melt</span>(SignTaxa<span class="op">$</span>TotalUp1)</a>
<a class="sourceLine" id="cb54-4" data-line-number="4"><span class="kw">ggplot</span>(<span class="dt">data =</span> melted_Table, <span class="kw">aes</span>(<span class="dt">x =</span> <span class="kw">reorder</span>(<span class="kw">rownames</span>(melted_Table), </a>
<a class="sourceLine" id="cb54-5" data-line-number="5">    <span class="op">-</span>value), <span class="dt">y =</span> value)) <span class="op">+</span><span class="st"> </span><span class="kw">scale_fill_gradient</span>(<span class="dt">low =</span> <span class="st">&quot;steelblue&quot;</span>, </a>
<a class="sourceLine" id="cb54-6" data-line-number="6">    <span class="dt">high =</span> <span class="st">&quot;#FF0066&quot;</span>, <span class="dt">na.value =</span> <span class="st">&quot;steelblue&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_classic</span>() <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb54-7" data-line-number="7"><span class="st">    </span><span class="kw">geom_bar</span>(<span class="dt">stat =</span> <span class="st">&quot;identity&quot;</span>, <span class="dt">fill =</span> <span class="st">&quot;#56B4E9&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb54-8" data-line-number="8"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.ticks =</span> <span class="kw">element_blank</span>(), <span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>(), </a>
<a class="sourceLine" id="cb54-9" data-line-number="9">        <span class="dt">axis.title.y =</span> <span class="kw">element_blank</span>()) <span class="op">+</span><span class="st"> </span>ggpubr<span class="op">::</span><span class="kw">rotate_x_text</span>(<span class="dv">90</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb54-10" data-line-number="10"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="kw">rel</span>(<span class="fl">0.9</span>)))</a></code></pre></div>
<p><img src="LinkHD_files/figure-html/unnamed-chunk-25-1.png" width="672" /></p>
<p>We can explore the relationships between enriched genus/families and the Ruminotypes. In our example, it is reasonable to expect the highest differences between samples classified as ruminotypes C1 and C3. The following examples are focused on members of Ruminococcaceae, Christensenellaceae and Succinivibrionaceae family, but the user can extend it to other family or taxonomic levels:</p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb55-1" data-line-number="1">Rumin &lt;-<span class="st"> </span><span class="kw">which</span>(Ruminotypes<span class="op">$</span>Taxa_16S[Ruminotypes<span class="op">$</span>Taxa_16S<span class="op">$</span>OTUID <span class="op">%in%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb55-2" data-line-number="2"><span class="st">    </span>SelectedB_16S, ]<span class="op">$</span>Family <span class="op">==</span><span class="st"> &quot;Ruminococcaceae&quot;</span>)</a>
<a class="sourceLine" id="cb55-3" data-line-number="3">OTUs &lt;-<span class="st"> </span>Ruminotypes<span class="op">$</span>Taxa_16S[Ruminotypes<span class="op">$</span>Taxa_16S<span class="op">$</span>OTUID <span class="op">%in%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb55-4" data-line-number="4"><span class="st">    </span>SelectedB_16S, ][Rumin, ]<span class="op">$</span>OTUID</a>
<a class="sourceLine" id="cb55-5" data-line-number="5">Rumin_Tot &lt;-<span class="st"> </span>Dataset[[<span class="st">&quot;16_S&quot;</span>]][, <span class="kw">colnames</span>(Dataset[[<span class="st">&quot;16_S&quot;</span>]]) <span class="op">%in%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb55-6" data-line-number="6"><span class="st">    </span>OTUs]</a>
<a class="sourceLine" id="cb55-7" data-line-number="7">RuminSum &lt;-<span class="st"> </span><span class="kw">apply</span>(Rumin_Tot, <span class="dv">1</span>, sum)</a>
<a class="sourceLine" id="cb55-8" data-line-number="8"></a>
<a class="sourceLine" id="cb55-9" data-line-number="9">Total &lt;-<span class="st"> </span><span class="kw">data.frame</span>(RuminSum, <span class="dt">Group =</span> <span class="kw">as.factor</span>(<span class="kw">compromise_coords</span>(Output)<span class="op">$</span>fit.cluster))</a>
<a class="sourceLine" id="cb55-10" data-line-number="10">p &lt;-<span class="st"> </span><span class="kw">ggplot</span>(Total, <span class="kw">aes</span>(<span class="dt">x =</span> Group, <span class="dt">y =</span> RuminSum, <span class="dt">fill =</span> Group)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb55-11" data-line-number="11"><span class="st">    </span><span class="kw">geom_boxplot</span>() <span class="op">+</span><span class="st"> </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;Tomato&quot;</span>, </a>
<a class="sourceLine" id="cb55-12" data-line-number="12">    <span class="st">&quot;Orange&quot;</span>, <span class="st">&quot;DodgerBlue&quot;</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">theme_classic</span>() <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="kw">paste</span>(<span class="st">&quot;Differences at level&quot;</span>, </a>
<a class="sourceLine" id="cb55-13" data-line-number="13">    <span class="kw">rownames</span>(melted_Table)[<span class="dv">16</span>]))</a>
<a class="sourceLine" id="cb55-14" data-line-number="14">p</a></code></pre></div>
<p><img src="LinkHD_files/figure-html/unnamed-chunk-26-1.png" width="672" /></p>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb56-1" data-line-number="1">Christ &lt;-<span class="st"> </span><span class="kw">which</span>(Ruminotypes<span class="op">$</span>Taxa_16S[Ruminotypes<span class="op">$</span>Taxa_16S<span class="op">$</span>OTUID <span class="op">%in%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb56-2" data-line-number="2"><span class="st">    </span>SelectedB_16S, ]<span class="op">$</span>Family <span class="op">==</span><span class="st"> </span><span class="kw">rownames</span>(melted_Table)[<span class="dv">14</span>])</a>
<a class="sourceLine" id="cb56-3" data-line-number="3">OTUs &lt;-<span class="st"> </span>Ruminotypes<span class="op">$</span>Taxa_16S[Ruminotypes<span class="op">$</span>Taxa_16S<span class="op">$</span>OTUID <span class="op">%in%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb56-4" data-line-number="4"><span class="st">    </span>SelectedB_16S, ][Christ, ]<span class="op">$</span>OTUID</a>
<a class="sourceLine" id="cb56-5" data-line-number="5">Christ_Tot &lt;-<span class="st"> </span>Dataset[[<span class="st">&quot;16_S&quot;</span>]][, <span class="kw">colnames</span>(Dataset[[<span class="st">&quot;16_S&quot;</span>]]) <span class="op">%in%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb56-6" data-line-number="6"><span class="st">    </span>OTUs]</a>
<a class="sourceLine" id="cb56-7" data-line-number="7">ChristSum &lt;-<span class="st"> </span><span class="kw">apply</span>(Christ_Tot, <span class="dv">1</span>, sum)</a>
<a class="sourceLine" id="cb56-8" data-line-number="8">Total &lt;-<span class="st"> </span><span class="kw">data.frame</span>(ChristSum, <span class="dt">Group =</span> <span class="kw">as.factor</span>(<span class="kw">compromise_coords</span>(Output)<span class="op">$</span>fit.cluster))</a>
<a class="sourceLine" id="cb56-9" data-line-number="9">p &lt;-<span class="st"> </span><span class="kw">ggplot</span>(Total, <span class="kw">aes</span>(<span class="dt">x =</span> Group, <span class="dt">y =</span> ChristSum, <span class="dt">fill =</span> Group)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb56-10" data-line-number="10"><span class="st">    </span><span class="kw">geom_boxplot</span>() <span class="op">+</span><span class="st"> </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;Tomato&quot;</span>, </a>
<a class="sourceLine" id="cb56-11" data-line-number="11">    <span class="st">&quot;Orange&quot;</span>, <span class="st">&quot;DodgerBlue&quot;</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">theme_classic</span>() <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="kw">paste</span>(<span class="st">&quot;Differences at level&quot;</span>, </a>
<a class="sourceLine" id="cb56-12" data-line-number="12">    <span class="kw">rownames</span>(melted_Table)[<span class="dv">14</span>]))</a>
<a class="sourceLine" id="cb56-13" data-line-number="13">p</a></code></pre></div>
<p><img src="LinkHD_files/figure-html/unnamed-chunk-27-1.png" width="672" /></p>
<p>The following code gives an alternative view to the previous plot:</p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb57-1" data-line-number="1"><span class="kw">ggplot</span>(Total, <span class="kw">aes</span>(<span class="dt">x =</span> Group, <span class="dt">y =</span> <span class="kw">rownames</span>(Total), <span class="dt">fill =</span> ChristSum)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb57-2" data-line-number="2"><span class="st">    </span><span class="kw">geom_tile</span>() <span class="op">+</span><span class="st"> </span><span class="kw">scale_fill_gradient</span>(<span class="dt">low =</span> <span class="st">&quot;steelblue&quot;</span>, </a>
<a class="sourceLine" id="cb57-3" data-line-number="3">    <span class="dt">high =</span> <span class="st">&quot;#FF0066&quot;</span>, <span class="dt">na.value =</span> <span class="st">&quot;steelblue&quot;</span>) <span class="op">+</span><span class="st"> </span>ggpubr<span class="op">::</span><span class="kw">rotate_x_text</span>(<span class="dv">45</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb57-4" data-line-number="4"><span class="st">    </span><span class="kw">theme_classic</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.ticks =</span> <span class="kw">element_blank</span>(), </a>
<a class="sourceLine" id="cb57-5" data-line-number="5">    <span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>(), <span class="dt">axis.title.y =</span> <span class="kw">element_blank</span>()) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb57-6" data-line-number="6"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.text.y =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="kw">rel</span>(<span class="fl">0.6</span>)))</a></code></pre></div>
<p><img src="LinkHD_files/figure-html/unnamed-chunk-28-1.png" width="672" /></p>
<p>Samples from the cluster 2 have a higher frequency of OTUs members of the family Christensenellaceae.</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb58-1" data-line-number="1">Succi &lt;-<span class="st"> </span><span class="kw">which</span>(Ruminotypes<span class="op">$</span>Taxa_16S[Ruminotypes<span class="op">$</span>Taxa_16S<span class="op">$</span>OTUID <span class="op">%in%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb58-2" data-line-number="2"><span class="st">    </span>SelectedB_16S, ]<span class="op">$</span>Family <span class="op">==</span><span class="st"> &quot;Succinivibrionaceae&quot;</span>)</a>
<a class="sourceLine" id="cb58-3" data-line-number="3">OTUs &lt;-<span class="st"> </span>Ruminotypes<span class="op">$</span>Taxa_16S[Ruminotypes<span class="op">$</span>Taxa_16S<span class="op">$</span>OTUID <span class="op">%in%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb58-4" data-line-number="4"><span class="st">    </span>SelectedB_16S, ][Succi, ]<span class="op">$</span>OTUID</a>
<a class="sourceLine" id="cb58-5" data-line-number="5">Succi_Tot &lt;-<span class="st"> </span>Dataset[[<span class="st">&quot;16_S&quot;</span>]][, <span class="kw">colnames</span>(Dataset[[<span class="st">&quot;16_S&quot;</span>]]) <span class="op">%in%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb58-6" data-line-number="6"><span class="st">    </span>OTUs]</a>
<a class="sourceLine" id="cb58-7" data-line-number="7">SucciSum &lt;-<span class="st"> </span><span class="kw">apply</span>(Succi_Tot, <span class="dv">1</span>, sum)</a>
<a class="sourceLine" id="cb58-8" data-line-number="8"></a>
<a class="sourceLine" id="cb58-9" data-line-number="9">Total &lt;-<span class="st"> </span><span class="kw">data.frame</span>(SucciSum, <span class="dt">Group =</span> <span class="kw">as.factor</span>(<span class="kw">compromise_coords</span>(Output)<span class="op">$</span>fit.cluster))</a>
<a class="sourceLine" id="cb58-10" data-line-number="10">Total &lt;-<span class="st"> </span>Total[<span class="op">-</span><span class="kw">which</span>(Total[, <span class="dv">1</span>] <span class="op">&gt;</span><span class="st"> </span><span class="dv">1</span>), ]</a>
<a class="sourceLine" id="cb58-11" data-line-number="11">p &lt;-<span class="st"> </span><span class="kw">ggplot</span>(Total, <span class="kw">aes</span>(<span class="dt">x =</span> Group, <span class="dt">y =</span> SucciSum, <span class="dt">fill =</span> Group)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb58-12" data-line-number="12"><span class="st">    </span><span class="kw">geom_boxplot</span>() <span class="op">+</span><span class="st"> </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;#e46563&quot;</span>, </a>
<a class="sourceLine" id="cb58-13" data-line-number="13">    <span class="st">&quot;#357fe3&quot;</span>, <span class="st">&quot;#75e463&quot;</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">theme_classic</span>() <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="kw">paste</span>(<span class="st">&quot;Differences at level&quot;</span>, </a>
<a class="sourceLine" id="cb58-14" data-line-number="14">    <span class="st">&quot;Succinivibrionaceae&quot;</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="kw">rel</span>(<span class="fl">1.5</span>)), </a>
<a class="sourceLine" id="cb58-15" data-line-number="15">    <span class="dt">legend.title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="kw">rel</span>(<span class="fl">1.5</span>)), <span class="dt">legend.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="kw">rel</span>(<span class="fl">1.5</span>)))</a>
<a class="sourceLine" id="cb58-16" data-line-number="16">p</a></code></pre></div>
<p><img src="LinkHD_files/figure-html/unnamed-chunk-29-1.png" width="672" /></p>
<p>As previous studies <span class="citation">(Kittelmann et al. 2013; Danielsson et al. 2017; Ramayo-Caldas et al. 2019)</span>, Succinivibrionaceae family is more prevalent on the lower CH4 emitting cows. The three most relevant families associated with Ch4y emission are Ruminococcaceae, Anaerolineaceae and Succinivibrionaceae.</p>
<p>Focusing on the protozoa dataset, the next figure represents the most important (enriched) Genus from variables selected by the regression biplot:</p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb59-1" data-line-number="1">SignTaxa &lt;-<span class="st"> </span><span class="kw">OTU2Taxa</span>(<span class="dt">Selection =</span> <span class="kw">VarTable</span>(Select_Var), </a>
<a class="sourceLine" id="cb59-2" data-line-number="2">    <span class="dt">TaxonInfo =</span> Ruminotypes<span class="op">$</span>Taxa_18S, <span class="dt">tableName =</span> <span class="st">&quot;18_S&quot;</span>)</a>
<a class="sourceLine" id="cb59-3" data-line-number="3">melted_Table &lt;-<span class="st"> </span><span class="kw">melt</span>(SignTaxa<span class="op">$</span>TotalUp1)</a>
<a class="sourceLine" id="cb59-4" data-line-number="4"><span class="kw">ggplot</span>(<span class="dt">data =</span> melted_Table, <span class="kw">aes</span>(<span class="dt">x =</span> <span class="kw">reorder</span>(<span class="kw">rownames</span>(melted_Table), </a>
<a class="sourceLine" id="cb59-5" data-line-number="5">    <span class="op">-</span>value), <span class="dt">y =</span> value)) <span class="op">+</span><span class="st"> </span><span class="kw">scale_fill_gradient</span>(<span class="dt">low =</span> <span class="st">&quot;steelblue&quot;</span>, </a>
<a class="sourceLine" id="cb59-6" data-line-number="6">    <span class="dt">high =</span> <span class="st">&quot;#B848DB&quot;</span>, <span class="dt">na.value =</span> <span class="st">&quot;steelblue&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">geom_bar</span>(<span class="dt">stat =</span> <span class="st">&quot;identity&quot;</span>, </a>
<a class="sourceLine" id="cb59-7" data-line-number="7">    <span class="dt">fill =</span> <span class="st">&quot;#B848DB&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_classic</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.ticks =</span> <span class="kw">element_blank</span>(), </a>
<a class="sourceLine" id="cb59-8" data-line-number="8">    <span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>(), <span class="dt">axis.title.y =</span> <span class="kw">element_blank</span>()) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb59-9" data-line-number="9"><span class="st">    </span>ggpubr<span class="op">::</span><span class="kw">rotate_x_text</span>(<span class="dv">90</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="kw">rel</span>(<span class="fl">1.5</span>)))</a></code></pre></div>
<p><img src="LinkHD_files/figure-html/unnamed-chunk-30-1.png" width="672" /></p>
<p>The most relevant genera at 18S rRNA gene level from variables selected by the differential abundance were:</p>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb60-1" data-line-number="1">SignTaxa &lt;-<span class="st"> </span><span class="kw">OTU2Taxa</span>(<span class="dt">Selection =</span> M, <span class="dt">TaxonInfo =</span> Ruminotypes<span class="op">$</span>Taxa_18S, </a>
<a class="sourceLine" id="cb60-2" data-line-number="2">    <span class="dt">tableName =</span> <span class="st">&quot;18_S&quot;</span>)</a>
<a class="sourceLine" id="cb60-3" data-line-number="3">melted_Table &lt;-<span class="st"> </span><span class="kw">melt</span>(SignTaxa<span class="op">$</span>TotalUp1[SignTaxa<span class="op">$</span>TotalUp1 <span class="op">&gt;</span><span class="st"> </span></a>
<a class="sourceLine" id="cb60-4" data-line-number="4"><span class="st">    </span><span class="dv">1</span>])</a>
<a class="sourceLine" id="cb60-5" data-line-number="5"><span class="kw">ggplot</span>(<span class="dt">data =</span> melted_Table, <span class="kw">aes</span>(<span class="dt">x =</span> <span class="kw">reorder</span>(<span class="kw">rownames</span>(melted_Table), </a>
<a class="sourceLine" id="cb60-6" data-line-number="6">    <span class="op">-</span>value), <span class="dt">y =</span> value)) <span class="op">+</span><span class="st"> </span><span class="kw">scale_fill_gradient</span>(<span class="dt">low =</span> <span class="st">&quot;steelblue&quot;</span>, </a>
<a class="sourceLine" id="cb60-7" data-line-number="7">    <span class="dt">high =</span> <span class="st">&quot;#FF0066&quot;</span>, <span class="dt">na.value =</span> <span class="st">&quot;steelblue&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">geom_bar</span>(<span class="dt">stat =</span> <span class="st">&quot;identity&quot;</span>, </a>
<a class="sourceLine" id="cb60-8" data-line-number="8">    <span class="dt">fill =</span> <span class="st">&quot;#56B4E9&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_classic</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.ticks =</span> <span class="kw">element_blank</span>(), </a>
<a class="sourceLine" id="cb60-9" data-line-number="9">    <span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>(), <span class="dt">axis.title.y =</span> <span class="kw">element_blank</span>()) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb60-10" data-line-number="10"><span class="st">    </span>ggpubr<span class="op">::</span><span class="kw">rotate_x_text</span>(<span class="dv">90</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="kw">rel</span>(<span class="fl">0.8</span>)))</a></code></pre></div>
<p><img src="LinkHD_files/figure-html/unnamed-chunk-31-1.png" width="672" /></p>
<p>In the next boxplot we represent the cluster distribution of the Diplodinim genus relative abundance.</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb61-1" data-line-number="1">Dip &lt;-<span class="st"> </span><span class="kw">which</span>(Ruminotypes<span class="op">$</span>Taxa_18S[Ruminotypes<span class="op">$</span>Taxa_18S<span class="op">$</span>OTUID <span class="op">%in%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb61-2" data-line-number="2"><span class="st">    </span>SelectedB_18S, ]<span class="op">$</span>Genus <span class="op">==</span><span class="st"> &quot;Diplodinium&quot;</span>)</a>
<a class="sourceLine" id="cb61-3" data-line-number="3">OTUs &lt;-<span class="st"> </span>Ruminotypes<span class="op">$</span>Taxa_18S[Ruminotypes<span class="op">$</span>Taxa_18S<span class="op">$</span>OTUID <span class="op">%in%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb61-4" data-line-number="4"><span class="st">    </span>SelectedB_18S, ][Dip, ]<span class="op">$</span>OTUID</a>
<a class="sourceLine" id="cb61-5" data-line-number="5">Dip_Tot &lt;-<span class="st"> </span>Dataset[[<span class="st">&quot;18_S&quot;</span>]][, <span class="kw">colnames</span>(Dataset[[<span class="st">&quot;18_S&quot;</span>]]) <span class="op">%in%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb61-6" data-line-number="6"><span class="st">    </span>OTUs]</a>
<a class="sourceLine" id="cb61-7" data-line-number="7">DipSum &lt;-<span class="st"> </span><span class="kw">apply</span>(Dip_Tot, <span class="dv">1</span>, sum)</a>
<a class="sourceLine" id="cb61-8" data-line-number="8"></a>
<a class="sourceLine" id="cb61-9" data-line-number="9">Total &lt;-<span class="st"> </span><span class="kw">data.frame</span>(DipSum, <span class="dt">Group =</span> <span class="kw">as.factor</span>(<span class="kw">compromise_coords</span>(Output)<span class="op">$</span>fit.cluster))</a>
<a class="sourceLine" id="cb61-10" data-line-number="10"></a>
<a class="sourceLine" id="cb61-11" data-line-number="11"></a>
<a class="sourceLine" id="cb61-12" data-line-number="12">p &lt;-<span class="st"> </span><span class="kw">ggplot</span>(Total, <span class="kw">aes</span>(<span class="dt">x =</span> Group, <span class="dt">y =</span> DipSum, <span class="dt">fill =</span> Group)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb61-13" data-line-number="13"><span class="st">    </span><span class="kw">geom_boxplot</span>() <span class="op">+</span><span class="st"> </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;Tomato&quot;</span>, </a>
<a class="sourceLine" id="cb61-14" data-line-number="14">    <span class="st">&quot;Orange&quot;</span>, <span class="st">&quot;DodgerBlue&quot;</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">theme_classic</span>() <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="kw">paste</span>(<span class="st">&quot;Differences at level&quot;</span>, </a>
<a class="sourceLine" id="cb61-15" data-line-number="15">    <span class="st">&quot;Diplodinium&quot;</span>))</a>
<a class="sourceLine" id="cb61-16" data-line-number="16">p</a></code></pre></div>
<p><img src="LinkHD_files/figure-html/unnamed-chunk-32-1.png" width="672" /></p>
<p>An alternative visualization of the previous result:</p>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb62-1" data-line-number="1"><span class="kw">ggplot</span>(Total, <span class="kw">aes</span>(<span class="dt">x =</span> Group, <span class="dt">y =</span> <span class="kw">rownames</span>(Total), <span class="dt">fill =</span> DipSum)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb62-2" data-line-number="2"><span class="st">    </span><span class="kw">geom_tile</span>() <span class="op">+</span><span class="st"> </span><span class="kw">scale_fill_gradient</span>(<span class="dt">low =</span> <span class="st">&quot;steelblue&quot;</span>, </a>
<a class="sourceLine" id="cb62-3" data-line-number="3">    <span class="dt">high =</span> <span class="st">&quot;#FF0066&quot;</span>, <span class="dt">na.value =</span> <span class="st">&quot;steelblue&quot;</span>) <span class="op">+</span><span class="st"> </span>ggpubr<span class="op">::</span><span class="kw">rotate_x_text</span>(<span class="dv">45</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb62-4" data-line-number="4"><span class="st">    </span><span class="kw">theme_classic</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.ticks =</span> <span class="kw">element_blank</span>(), </a>
<a class="sourceLine" id="cb62-5" data-line-number="5">    <span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>(), <span class="dt">axis.title.y =</span> <span class="kw">element_blank</span>()) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb62-6" data-line-number="6"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.text.y =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="kw">rel</span>(<span class="fl">0.6</span>)))</a></code></pre></div>
<p><img src="LinkHD_files/figure-html/unnamed-chunk-33-1.png" width="672" /></p>
<p>The top relevant genera were the same for both approaches: Diplodinium, Ophryoscolex, Tetratrichomonas, Tritrichomonas.</p>
<p>The enriched genus/families at Archaea level:</p>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb63-1" data-line-number="1">SignTaxa &lt;-<span class="st"> </span><span class="kw">OTU2Taxa</span>(<span class="dt">Selection =</span> <span class="kw">VarTable</span>(Select_Var), </a>
<a class="sourceLine" id="cb63-2" data-line-number="2">    <span class="dt">TaxonInfo =</span> Ruminotypes<span class="op">$</span>Taxa_Archaea, <span class="dt">tableName =</span> <span class="st">&quot;Archaea&quot;</span>)</a>
<a class="sourceLine" id="cb63-3" data-line-number="3">melted_Table &lt;-<span class="st"> </span><span class="kw">melt</span>(SignTaxa<span class="op">$</span>TotalUp1)</a>
<a class="sourceLine" id="cb63-4" data-line-number="4"><span class="kw">ggplot</span>(<span class="dt">data =</span> melted_Table, <span class="kw">aes</span>(<span class="dt">x =</span> <span class="kw">reorder</span>(<span class="kw">rownames</span>(melted_Table), </a>
<a class="sourceLine" id="cb63-5" data-line-number="5">    <span class="op">-</span>value), <span class="dt">y =</span> value)) <span class="op">+</span><span class="st"> </span><span class="kw">scale_fill_gradient</span>(<span class="dt">low =</span> <span class="st">&quot;steelblue&quot;</span>, </a>
<a class="sourceLine" id="cb63-6" data-line-number="6">    <span class="dt">high =</span> <span class="st">&quot;#46D6AD&quot;</span>, <span class="dt">na.value =</span> <span class="st">&quot;steelblue&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">geom_bar</span>(<span class="dt">stat =</span> <span class="st">&quot;identity&quot;</span>, </a>
<a class="sourceLine" id="cb63-7" data-line-number="7">    <span class="dt">fill =</span> <span class="st">&quot;#46D6AD&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_classic</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.ticks =</span> <span class="kw">element_blank</span>(), </a>
<a class="sourceLine" id="cb63-8" data-line-number="8">    <span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>(), <span class="dt">axis.title.y =</span> <span class="kw">element_blank</span>()) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb63-9" data-line-number="9"><span class="st">    </span>ggpubr<span class="op">::</span><span class="kw">rotate_x_text</span>(<span class="dv">90</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="kw">rel</span>(<span class="fl">1.5</span>)))</a></code></pre></div>
<p><img src="LinkHD_files/figure-html/unnamed-chunk-34-1.png" width="672" /></p>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb64-1" data-line-number="1">SignTaxa &lt;-<span class="st"> </span><span class="kw">OTU2Taxa</span>(<span class="dt">Selection =</span> M, <span class="dt">TaxonInfo =</span> Ruminotypes<span class="op">$</span>Taxa_Archaea, </a>
<a class="sourceLine" id="cb64-2" data-line-number="2">    <span class="dt">tableName =</span> <span class="st">&quot;Archaea&quot;</span>)</a>
<a class="sourceLine" id="cb64-3" data-line-number="3">melted_Table &lt;-<span class="st"> </span><span class="kw">melt</span>(SignTaxa<span class="op">$</span>TotalUp1)</a>
<a class="sourceLine" id="cb64-4" data-line-number="4"><span class="kw">ggplot</span>(<span class="dt">data =</span> melted_Table, <span class="kw">aes</span>(<span class="dt">x =</span> <span class="kw">reorder</span>(<span class="kw">rownames</span>(melted_Table), </a>
<a class="sourceLine" id="cb64-5" data-line-number="5">    <span class="op">-</span>value), <span class="dt">y =</span> value)) <span class="op">+</span><span class="st"> </span><span class="kw">scale_fill_gradient</span>(<span class="dt">low =</span> <span class="st">&quot;steelblue&quot;</span>, </a>
<a class="sourceLine" id="cb64-6" data-line-number="6">    <span class="dt">high =</span> <span class="st">&quot;#FF0066&quot;</span>, <span class="dt">na.value =</span> <span class="st">&quot;steelblue&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">geom_bar</span>(<span class="dt">stat =</span> <span class="st">&quot;identity&quot;</span>, </a>
<a class="sourceLine" id="cb64-7" data-line-number="7">    <span class="dt">fill =</span> <span class="st">&quot;#56B4E9&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_classic</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.ticks =</span> <span class="kw">element_blank</span>(), </a>
<a class="sourceLine" id="cb64-8" data-line-number="8">    <span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>(), <span class="dt">axis.title.y =</span> <span class="kw">element_blank</span>()) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb64-9" data-line-number="9"><span class="st">    </span>ggpubr<span class="op">::</span><span class="kw">rotate_x_text</span>(<span class="dv">90</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="kw">rel</span>(<span class="fl">0.9</span>)))</a></code></pre></div>
<p><img src="LinkHD_files/figure-html/unnamed-chunk-35-1.png" width="672" /></p>
<p>Differences in the relative abundance of <em>Methanobrevibacter</em> between the clusters of samples are represented in the next figure. As expected, the cluster of the lower emitting samples has lower relative abundance of Methanobrevibacter.</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb65-1" data-line-number="1">Met &lt;-<span class="st"> </span><span class="kw">which</span>(Ruminotypes<span class="op">$</span>Taxa_Archaea[Ruminotypes<span class="op">$</span>Taxa_Archaea<span class="op">$</span>OTU <span class="op">%in%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb65-2" data-line-number="2"><span class="st">    </span>SelectedB_Archaea, ]<span class="op">$</span>Genus <span class="op">==</span><span class="st"> </span><span class="kw">levels</span>(Ruminotypes<span class="op">$</span>Taxa_Archaea<span class="op">$</span>Genus)[<span class="dv">4</span>])</a>
<a class="sourceLine" id="cb65-3" data-line-number="3">OTUs &lt;-<span class="st"> </span>Ruminotypes<span class="op">$</span>Taxa_Archaea[Ruminotypes<span class="op">$</span>Taxa_Archaea<span class="op">$</span>OTU <span class="op">%in%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb65-4" data-line-number="4"><span class="st">    </span>SelectedB_Archaea, ][Met, ]<span class="op">$</span>OTU</a>
<a class="sourceLine" id="cb65-5" data-line-number="5">Met_Tot &lt;-<span class="st"> </span>Dataset[[<span class="st">&quot;Archaea&quot;</span>]][, <span class="kw">colnames</span>(Dataset[[<span class="st">&quot;Archaea&quot;</span>]]) <span class="op">%in%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb65-6" data-line-number="6"><span class="st">    </span>OTUs]</a>
<a class="sourceLine" id="cb65-7" data-line-number="7">MetSum &lt;-<span class="st"> </span><span class="kw">apply</span>(Met_Tot, <span class="dv">1</span>, sum)</a>
<a class="sourceLine" id="cb65-8" data-line-number="8">Total &lt;-<span class="st"> </span><span class="kw">data.frame</span>(MetSum, <span class="dt">Group =</span> <span class="kw">as.factor</span>(<span class="kw">compromise_coords</span>(Output)<span class="op">$</span>fit.cluster))</a>
<a class="sourceLine" id="cb65-9" data-line-number="9">p &lt;-<span class="st"> </span><span class="kw">ggplot</span>(Total, <span class="kw">aes</span>(<span class="dt">x =</span> Group, <span class="dt">y =</span> MetSum, <span class="dt">fill =</span> Group)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb65-10" data-line-number="10"><span class="st">    </span><span class="kw">geom_boxplot</span>() <span class="op">+</span><span class="st"> </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">c</span>(<span class="st">&quot;Tomato&quot;</span>, </a>
<a class="sourceLine" id="cb65-11" data-line-number="11">    <span class="st">&quot;Orange&quot;</span>, <span class="st">&quot;DodgerBlue&quot;</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">theme_classic</span>() <span class="op">+</span><span class="st"> </span><span class="kw">ggtitle</span>(<span class="kw">paste</span>(<span class="st">&quot;Differences at level&quot;</span>, </a>
<a class="sourceLine" id="cb65-12" data-line-number="12">    <span class="st">&quot;Methanobrevibacter&quot;</span>))</a>
<a class="sourceLine" id="cb65-13" data-line-number="13">p</a></code></pre></div>
<p><img src="LinkHD_files/figure-html/unnamed-chunk-36-1.png" width="672" /></p>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb66-1" data-line-number="1"><span class="kw">ggplot</span>(Total, <span class="kw">aes</span>(<span class="dt">x =</span> Group, <span class="dt">y =</span> <span class="kw">rownames</span>(Total), <span class="dt">fill =</span> MetSum)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb66-2" data-line-number="2"><span class="st">    </span><span class="kw">geom_tile</span>() <span class="op">+</span><span class="st"> </span><span class="kw">scale_fill_gradient</span>(<span class="dt">low =</span> <span class="st">&quot;steelblue&quot;</span>, </a>
<a class="sourceLine" id="cb66-3" data-line-number="3">    <span class="dt">high =</span> <span class="st">&quot;#FF0066&quot;</span>, <span class="dt">na.value =</span> <span class="st">&quot;steelblue&quot;</span>) <span class="op">+</span><span class="st"> </span>ggpubr<span class="op">::</span><span class="kw">rotate_x_text</span>(<span class="dv">45</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb66-4" data-line-number="4"><span class="st">    </span><span class="kw">theme_classic</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.ticks =</span> <span class="kw">element_blank</span>(), </a>
<a class="sourceLine" id="cb66-5" data-line-number="5">    <span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>(), <span class="dt">axis.title.y =</span> <span class="kw">element_blank</span>()) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb66-6" data-line-number="6"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.text.y =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="kw">rel</span>(<span class="fl">0.6</span>)))</a></code></pre></div>
<p><img src="LinkHD_files/figure-html/unnamed-chunk-37-1.png" width="672" /></p>
</div>
<div id="Taxon" class="section level2 unnumbered">
<h2>Taxon set enrichment analysis from a user-defined OTUs’ list</h2>
<p>OTU2Taxa function can be used to aggregate OTUs into their taxonomic characteristics and to compute the most significant (“enrichment”) genera/families to any OTUs’ list, i.e., a list obtained by another methodology. For instance, in <span class="citation">(Ramayo-Caldas et al. 2019)</span> the authors selected two OTUs list by applying SPLS and Discriminant SPLS (SPLS DA). Here, we show how to use these external dataset with OTU2Taxa:</p>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb67-1" data-line-number="1"><span class="co"># library(&#39;readxl&#39;) library(&#39;XLConnect&#39;)</span></a>
<a class="sourceLine" id="cb67-2" data-line-number="2"><span class="co"># library(&#39;openxlsx&#39;)</span></a>
<a class="sourceLine" id="cb67-3" data-line-number="3"></a>
<a class="sourceLine" id="cb67-4" data-line-number="4"><span class="co"># Download the list from supplementary material:</span></a>
<a class="sourceLine" id="cb67-5" data-line-number="5"><span class="co"># temp = tempfile(fileext = &#39;.xls&#39;) dataURL &lt;-</span></a>
<a class="sourceLine" id="cb67-6" data-line-number="6"><span class="co"># &#39;https://onlinelibrary.wiley.com/action/downloadSupplement?doi=10.1111%2Fjbg.12427&amp;file=jbg12427-sup-0004-TableS4.xls&#39;</span></a>
<a class="sourceLine" id="cb67-7" data-line-number="7"><span class="co"># download.file(dataURL, destfile=temp, mode=&#39;wb&#39;)</span></a>
<a class="sourceLine" id="cb67-8" data-line-number="8"><span class="co"># SPLS_DA&lt;-read_excel(temp,sheet=1)</span></a>
<a class="sourceLine" id="cb67-9" data-line-number="9"><span class="co"># SPLS&lt;-read_excel(temp,sheet=2)</span></a>
<a class="sourceLine" id="cb67-10" data-line-number="10"></a>
<a class="sourceLine" id="cb67-11" data-line-number="11"><span class="co"># Create a list with selected OTU. Note that if you</span></a>
<a class="sourceLine" id="cb67-12" data-line-number="12"><span class="co"># have only one table, the list should have lenght</span></a>
<a class="sourceLine" id="cb67-13" data-line-number="13"><span class="co"># equal to 1.</span></a>
<a class="sourceLine" id="cb67-14" data-line-number="14"></a>
<a class="sourceLine" id="cb67-15" data-line-number="15"><span class="co"># Selected_OTU&lt;-list(&#39;SPLS&#39;=SPLS$OTU,&#39;SPLS_DA&#39;=SPLS_DA$OTU)</span></a>
<a class="sourceLine" id="cb67-16" data-line-number="16"><span class="co"># OTUs have to be the lists names:</span></a>
<a class="sourceLine" id="cb67-17" data-line-number="17"><span class="co"># names(Selected_OTU$SPLS)&lt;-SPLS$OTU</span></a>
<a class="sourceLine" id="cb67-18" data-line-number="18"><span class="co"># names(Selected_OTU$SPLS_DA)&lt;-SPLS_DA$OTU</span></a>
<a class="sourceLine" id="cb67-19" data-line-number="19"></a>
<a class="sourceLine" id="cb67-20" data-line-number="20"></a>
<a class="sourceLine" id="cb67-21" data-line-number="21"><span class="co"># TaxonInfo is the same than 16_S taxon info into</span></a>
<a class="sourceLine" id="cb67-22" data-line-number="22"><span class="co"># Ruminotypes dataset: Apply OTU2Taxa</span></a>
<a class="sourceLine" id="cb67-23" data-line-number="23"><span class="co"># SignTaxa&lt;-OTU2Taxa(Selection=Selected_OTU,TaxonInfo=Ruminotypes$Taxa_16S,tableName=&#39;SPLS&#39;,AnalysisLev</span></a>
<a class="sourceLine" id="cb67-24" data-line-number="24"><span class="co"># = &#39;Family&#39;)</span></a>
<a class="sourceLine" id="cb67-25" data-line-number="25"><span class="co"># SignTaxa2&lt;-OTU2Taxa(Selection=Selected_OTU,TaxonInfo=Ruminotypes$Taxa_16S,tableName=&#39;SPLS_DA&#39;,AnalysisLev</span></a>
<a class="sourceLine" id="cb67-26" data-line-number="26"><span class="co"># = &#39;Family&#39;)</span></a>
<a class="sourceLine" id="cb67-27" data-line-number="27"></a>
<a class="sourceLine" id="cb67-28" data-line-number="28"></a>
<a class="sourceLine" id="cb67-29" data-line-number="29"></a>
<a class="sourceLine" id="cb67-30" data-line-number="30"><span class="co"># melted_Table &lt;- melt(SignTaxa2$TotalUp1)</span></a>
<a class="sourceLine" id="cb67-31" data-line-number="31"><span class="co"># ggplot(data =melted_Table, aes(x=</span></a>
<a class="sourceLine" id="cb67-32" data-line-number="32"><span class="co"># reorder(rownames(melted_Table), -value),</span></a>
<a class="sourceLine" id="cb67-33" data-line-number="33"><span class="co"># y=value)) + scale_fill_gradient(low =</span></a>
<a class="sourceLine" id="cb67-34" data-line-number="34"><span class="co"># &#39;steelblue&#39;,high =</span></a>
<a class="sourceLine" id="cb67-35" data-line-number="35"><span class="co"># &#39;#FF0066&#39;,na.value=&#39;steelblue&#39;) +</span></a>
<a class="sourceLine" id="cb67-36" data-line-number="36"><span class="co"># theme_classic()+ geom_bar(stat=&#39;identity&#39;, fill=</span></a>
<a class="sourceLine" id="cb67-37" data-line-number="37"><span class="co"># &#39;#ff661a&#39; )+ theme( axis.ticks =</span></a>
<a class="sourceLine" id="cb67-38" data-line-number="38"><span class="co"># element_blank(),axis.title.x=element_blank(),axis.title.y</span></a>
<a class="sourceLine" id="cb67-39" data-line-number="39"><span class="co"># = element_blank())+ ggpubr::rotate_x_text(90) +</span></a>
<a class="sourceLine" id="cb67-40" data-line-number="40"><span class="co"># theme(axis.text.x=element_text(size=rel(0.9)))</span></a></code></pre></div>
</div>
<div id="Final" class="section level2 unnumbered">
<h2>Final Remarks</h2>
<p>In summary,Link-HD combines some well-known multivariate techniques to perform data integration with cluster, variable selection and network analysis. We show the utility of Link-HD to integrate heterogeneous datasets. Applied to data from the TARA Ocean expedition, Link-HD reproduces the relevance of members of Proteobacteria phyla and temperature on the functionality and stratification of this ecosystem. Regarding the rumen microbiota, the consensus space inferred by Link-HD shows the association of the structure of the rumen microbiome with <span class="math inline">\(CH_4\)</span> emission. The method also facilitates the study of the relationships between observations and features for performing variable selection. In this sense, a good performance of Link-HD with other available integrative methodologies, such as mixKernel was demonstrated.</p>
</div>
<div id="Appendix" class="section level2 unnumbered">
<h2>Appendix</h2>
<p>List of LinkHD dependencies:</p>
<pre><code>methods,
scales,
ggplot2,
cluster,
graphics, 
ggpubr, 
reshape2,
gridExtra,
vegan,
data.table,
rio, 
MultiAssayExperiment</code></pre>
</div>
<div id="References" class="section level2 unnumbered">
<h2>References</h2>
<div id="refs" class="references">
<div>
<p>Aitchison, John. 1982. “The Statistical Analysis of Compositional Data.” <em>Journal of the Royal Statistical Society: Series B (Methodological)</em> 44 (2). Wiley Online Library: 139–60.</p>
</div>
<div>
<p>Argelaguet, Ricard, Britta Velten, Damien Arnol, Sascha Dietrich, Thorsten Zenz, John C Marioni, Florian Buettner, Wolfgang Huber, and Oliver Stegle. 2018. “Multi-Omics Factor Analysis—a Framework for Unsupervised Integration of Multi-Omics Data Sets.” <em>Molecular Systems Biology</em> 14 (6). EMBO Press: e8124.</p>
</div>
<div>
<p>Benjamini, Yoav, and Yosef Hochberg. 1995. “Controlling the False Discovery Rate: A Practical and Powerful Approach to Multiple Testing.” <em>Journal of the Royal Statistical Society: Series B (Methodological)</em> 57 (1). Wiley Online Library: 289–300.</p>
</div>
<div>
<p>Danielsson, Rebecca, Johan Dicksved, Li Sun, Horacio Gonda, Bettina Müller, Anna Schnürer, and Jan Bertilsson. 2017. “Methane Production in Dairy Cows Correlates with Rumen Methanogenic and Bacterial Community Structure.” <em>Frontiers in Microbiology</em> 8. Frontiers: 226.</p>
</div>
<div>
<p>Escoufier, Y, P Cazes, and others. 1976. “Opérateurs et Analyse Des Tableaux à Plus de Deux Dimensions.” <em>Cahiers Du Bureau Universitaire de Recherche Opérationnelle</em> 25: 61–89.</p>
</div>
<div>
<p>Gabriel, K. R. 1971. “The Biplot Graphic Display of Matrices with Application to Principal Component Analysis.” <em>Biometrika</em> 58 (3): 453. <a href="https://doi.org/10.2307/2334381">https://doi.org/10.2307/2334381</a>.</p>
</div>
<div>
<p>Gower, John C, and David J Hand. 1995. <em>Biplots</em>. Vol. 54. CRC Press.</p>
</div>
<div>
<p>Greenacre, Michael J. 2010. <em>Biplots in Practice</em>. Fundacion BBVA.</p>
</div>
<div>
<p>Kittelmann, Sandra, Henning Seedorf, William A Walters, Jose C Clemente, Rob Knight, Jeffrey I Gordon, and Peter H Janssen. 2013. “Simultaneous Amplicon Sequencing to Explore Co-Occurrence Patterns of Bacterial, Archaeal and Eukaryotic Microorganisms in Rumen Microbial Communities.” <em>PloS One</em> 8 (2). Public Library of Science: e47879.</p>
</div>
<div>
<p>Kruskal, William H, and W Allen Wallis. 1952. “Use of Ranks in One-Criterion Variance Analysis.” <em>Journal of the American Statistical Association</em> 47 (260). Taylor &amp; Francis Group: 583–621.</p>
</div>
<div>
<p>Lavit, Christine. 1988. “Presentation de La Méthode Statis Permettant L’analyse Conjointe de Plusieurs Tableaux de Données Quantitatives.” <em>Cachiers de La Recherche Dévelopment</em> 18: 49–60.</p>
</div>
<div>
<p>Lavit, Christine, Yves Escoufier, Robert Sabatier, and Pierre Traissac. 1994. “The ACT (STATIS method).” <em>Computational Statistics &amp; Data Analysis</em> 18 (1): 97–119. <a href="https://doi.org/10.1016/0167-9473(94)90134-1">https://doi.org/10.1016/0167-9473(94)90134-1</a>.</p>
</div>
<div>
<p>Mariette, Jérôme, and Nathalie Villa-Vialaneix. 2017. “Unsupervised Multiple Kernel Learning for Heterogeneous Data Integration.” <em>Bioinformatics</em> 34 (6). Oxford University Press: 1009–15.</p>
</div>
<div>
<p>Paulson, Joseph N, O Colin Stine, Héctor Corrada Bravo, and Mihai Pop. 2013. “Differential Abundance Analysis for Microbial Marker-Gene Surveys.” <em>Nature Methods</em> 10 (12). Nature Publishing Group: 1200.</p>
</div>
<div>
<p>Plantes, Henri L’Hermier des. 1976. “Structuration Des Tableaux à Trois Indices de La Statistique: Théorie et Application d’une Méthode d’analyse Conjointe.” PhD thesis, Université des sciences et techniques du Languedoc.</p>
</div>
<div>
<p>Ramayo-Caldas, Yuliaxis, Laura Zingaretti, Milka Popova, Jordi Estellé, Aurelien Bernard, Nicolas Pons, Pau Bellot, et al. 2019. “Identification of Rumen Microbial Biomarkers Linked to Methane Emission in Holstein Dairy Cows.” <em>Journal of Animal Breeding and Genetics</em>. Wiley Online Library.</p>
</div>
<div>
<p>Robert, Paul, and Yves Escoufier. 1976. “A unifying tool for linear multivariate statistical methods: the RV-coefficient.” <em>Applied Statistics</em>. JSTOR, 257–65.</p>
</div>
<div>
<p>Rohart, Florian, Benoit Gautier, Amrit Singh, and Kim-Anh Lê Cao. 2017. “MixOmics: An R Package for ‘Omics Feature Selection and Multiple Data Integration.” <em>PLoS Computational Biology</em> 13 (11). Public Library of Science: e1005752.</p>
</div>
<div>
<p>Rousseeuw, Peter J. 1987. “Silhouettes: A Graphical Aid to the Interpretation and Validation of Cluster Analysis.” <em>Journal of Computational and Applied Mathematics</em> 20. Elsevier: 53–65.</p>
</div>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>


    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
},
"search": false
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
